docs/_docs/04_community/client-factory.md

```diff
@@ -0,0 +1,124 @@
+
+## Background
+
+The Azure REST APIs support both Public and sovereign cloud environments (at the moment this includes Public, US Government, Germany, China, and Azure Stack environments).  If you are interacting with an environment other than public cloud, you need to set the base URI for the Azure REST API you are interacting with.
+
+### Base URI
+
+You must use the correct base URI's for the Azure REST API's (either directly or via Azure SDK for GO) to communicate with a cloud environment other than Azure Public. The Azure Go SDK supports this by using the `WithBaseURI` suffixed calls when creating service clients. For example, when using the `VirtualMachinesClient` with the public cloud, a developer would normally write code for the public cloud like so:
+
+```golang
+import (
+    "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+)
+
+func SomeVMHelperMethod() {
+    subscriptionID := "your subscription ID"
+
+    // Create a VM client and return
+    vmClient, err := compute.NewVirtualMachinesClient(subscriptionID)
+
+    // Use client / etc
+}
+```
+
+However, this code will not work in non-Public cloud environments as the REST endpoints have different URIs depending on environment.  Instead, you need to use an alternative method (provided in the Azure REST SDK for Go) to get a properly configured client (*all REST API clients should support this alternate method*):
+
+```golang
+import (
+    "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+)
+
+func SomeVMHelperMethod() {
+    subscriptionID := "your subscription ID"
+    baseURI := "management.azure.com"
+
+    // Create a VM client and return
+    vmClient, err := compute.NewVirtualMachinesClientWithBaseURI(baseURI, subscriptionID)
+
+    // Use client / etc
+}
+```
+
+Using code similar to above, you can communicate with any Azure cloud environment just by changing the base URI that is passed to the clients (Azure Public shown in above example).
+
+## Lookup Environment Metadata
+
+Developers MUST avoid hardcoding these base URI's.  Instead, they should be looked up from an authoritative source. The AutoRest-GO library (used by the Go SDK) provides such functionality. The `client_factory` module makes use of the AutoRest `EnvironmentFromName(envName string)` function to return the appropriate structure.  This method and Environment structure is documented on GoDoc [here](https://godoc.org/github.com/Azure/go-autorest/autorest/azure#EnvironmentFromName).
+
+To configure different cloud environments, we will use the same `AZURE_ENVIRONMENT` environment variable that the Go SDK uses. This can currently be set to one of the following values:
+
+|Value                      |Cloud Environment  |
```

Comment:
Doh. Looks like we don't have table styling in the CSS for this repo. Or maybe we do and it's just not being applied for some reason? Prob worth filing a bug and we can come back to it later.

<img width="471" alt="Screen Shot 2020-11-17 at 5 18 34 PM" src="https://user-images.githubusercontent.com/711908/99423898-f73e7c80-28f8-11eb-8a94-244c9d7dc089.png">

Reply:
Added bug #722

Reply:
Thanks!